### PR TITLE
Miscellaneous fixes resulting from the work on PSA-Crypto

### PR DESCRIPTION
**Pull request by ronald-cron-arm**
## Description

Various fixes, I came across while working on the PSA cryptography repository.
Fixes https://github.com/Mbed-TLS/psa-crypto/issues/22

## PR checklist

Please tick as appropriate and edit the reasons (e.g.: "backport: not needed because this is a new feature")

- [x] **changelog** provided
- [x] **backport** does not seem to be required. The only relevant commit is e71c540ec (change of some Windows related guard in check_config.h) but not clear it is worth the effort.
- [x] **tests** non-regression test for the CCM* with no tag issue provided

### CODE REVIEW

**Comment by ronald-cron-arm** [comment]
The failure in Internal CI is a cloning error thus the CI can be considered as passed.

File: tests/suites/test_suite_psa_crypto.function [hunk 1]
```diff
@@ -8979,7 +8979,7 @@ exit:
 }
 /* END_CASE */
 
-/* BEGIN_CASE depends_on:PSA_WANT_ALG_SHA_256:MBEDTLS_PSA_BUILTIN_ALG_TLS12_PSK_TO_MS */
+/* BEGIN_CASE depends_on:PSA_WANT_ALG_SHA_256:MBEDTLS_PSA_BUILTIN_ALG_TLS12_ECJPAKE_TO_MS */
```
Comment (davidhorstmann-arm) [inline tests/suites/test_suite_psa_crypto.function:8982]: I think this should be `MBEDTLS_PSA_BUILTIN_ALG_TLS12_ECJPAKE_TO_PMS`, based on reading `include/mbedtls/config_psa.h`

File: library/psa_util.c [hunk 2]
```diff
@@ -28,9 +28,12 @@
 #include <mbedtls/psa_util.h>
 #include <mbedtls/error.h>
 #include <mbedtls/lms.h>
-#include <mbedtls/ssl.h>
 #include <mbedtls/rsa.h>
 
+#if defined(MBEDTLS_USE_PSA_CRYPTO) || defined(MBEDTLS_SSL_PROTO_TLS1_3)
```
Comment (davidhorstmann-arm) [inline library/psa_util.c:33]: Why do we need `ssl.h` if `USE_PSA_CRYPTO` is defined? Surely we might be doing crypto only, not TLS?

**Review by davidhorstmann-arm — CHANGES_REQUESTED** [review]
A few questions

File: tests/suites/test_suite_psa_crypto.function (hunk 1 above)
Comment (ronald-cron-arm) [reply tests/suites/test_suite_psa_crypto.function:8982]: Yes good catch

File: library/psa_util.c (hunk 2 above)
Comment (ronald-cron-arm) [reply library/psa_util.c:33]: Change removed as part of the rebase as it was addressed by abfe640864fc041e16f09356cf049a3c113532fa.

**Review by davidhorstmann-arm — DISMISSED** [review]
LGTM, thanks!

**Comment by ronald-cron-arm** [comment]
Rebased, on less commit following the merge of #7432

**Review by davidhorstmann-arm — DISMISSED** [review]
LGTM, thanks!

File: include/mbedtls/check_config.h [hunk 3]
```diff
@@ -35,7 +35,7 @@
 
 #include <stdint.h>
 
-#if defined(_WIN32)
+#if defined(__MINGW32__) || (defined(_MSC_VER) && _MSC_VER <= 1900)
```
[the remark this answers is not in the document]
Comment (mpg) [reply include/mbedtls/check_config.h:38]: @ronald-cron-arm  Yes, I think your reasoning is correct, and the guard here should evolve in the same way as the ones in `platform.h` have.

File: tests/scripts/all.sh [hunk 4]
```diff
@@ -3169,6 +3169,18 @@ component_test_psa_crypto_config_chachapoly_disabled() {
     make test
 }
 
+component_test_ccm_aes_sha256() {
+    msg "build: CCM + AES + SHA256 configuration"
+
+    cp tests/include/test/drivers/config_test_driver.h  include/mbedtls/mbedtls_config.h
+    cp configs/crypto-config-ccm-aes-sha256.h           include/psa/crypto_config.h
```
Comment (bensze01) [inline tests/scripts/all.sh:3176]: We should use `"$CONFIG_H"`, `"$CRYPTO_CONFIG_H"` and `"$CONFIG_TEST_DRIVER_H"` here

**Review by bensze01 — CHANGES_REQUESTED** [review]

**Comment by ronald-cron-arm** [comment]
The only CI error is a CI glitch thus CI can be considered as passed. @bensze01 I believe I have addressed your comments, please have another look.

**Review by bensze01 — APPROVED** [review]
Looks good to me.

**Review by davidhorstmann-arm — APPROVED** [review]
LGTM, thanks!

**Pull request merged**